### PR TITLE
Return Status on Providing Console GOP

### PR DESCRIPTION
**Pull request by dakanji**
Closes acidanthera/bugtracker#1135

### CODE REVIEW

File: Library/OcConsoleLib/ConsoleGop.c [hunk 1]
```diff
@@ -150,51 +150,52 @@ OcProvideConsoleGop (
 
     if (EFI_ERROR (Status)) {
       DEBUG ((DEBUG_INFO, "OCC: No handles with GOP protocol - %r\n", Status));
-      return;
```
Comment (vit9696) [inline Library/OcConsoleLib/ConsoleGop.c:153]: Do not change the control flow here and increase the indentation level. Return status here instead and update return calls below.

File: Library/OcConsoleLib/ConsoleGop.c [hunk 2]
```diff
@@ -192,8 +192,10 @@ OcProvideConsoleGop (
     }
 
     mConsoleGraphicsOutput = Gop;
+    return EFI_SUCCESS;
   } else {
```
Comment (vit9696) [inline Library/OcConsoleLib/ConsoleGop.c:196]: Drop the else, do not use else after `return`.

Comment (dakanji) [reply Library/OcConsoleLib/ConsoleGop.c:196]: A bit confused. Wouldn't deleting the else block result in some items in the flow (those that have an error status at the LocateProtocol command) get to the end without returning anything? 

Actually, the information about the missing compatible GOP at this stage is important (why I am looking at this in the first instance) and needed to return a different status to ID this condition.

Alternative is to move the EFI_SUCCESS to the end of the flow.

Thoughts?

Comment (vit9696) [reply Library/OcConsoleLib/ConsoleGop.c:196]: Not deleting the `else` block, but rewriting the control flow to:

```C
if (x) {
  …
  return EFI_SUCCESS;
}

DEBUG ((DEBUG_WARN, "OCC: Missing compatible GOP - %r\n", Status));
return EFI_NOT_FOUND;
```

It is considered bad practice to write an `else` case when the previous case does not return control. See more details [here](http://clang.llvm.org/extra/clang-tidy/checks/readability-else-after-return.html).

Comment (dakanji) [reply Library/OcConsoleLib/ConsoleGop.c:195]: Got it.

Anyway, already pushed a change that does not need this. Just returns status after the if/else block.

**Pull request merged**